File: RawatJalanApp/app/Models/rekapMedis.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class rekapMedis extends Model
{
    use HasFactory;

    protected $table = 'rekap_medis';

    protected $primaryKey = 'idRekapMedis'; 

    protected $fillable = [
        'idRekapMedis',
        'noRekapMedis',
        'idDokter',
        'tanggalKunjungan',
        'keluhan',
        'diagnosa',
        'tekananDarah',
        'suhuTubuh',
        'beratBadan',
        'tinggiBadan',
    ];

    public function pasien(): BelongsTo
    {
        return $this->belongsTo(Pasien::class, 'noRekapMedis');
    }

    public function dokter(): BelongsTo
    {
        return $this->belongsTo(Dokter::class, 'idDokter');
    }

}

File: rumahsakitapp/lib/model/rekap_medis.dart
class RekapMedisModel {
  final String tanggalKunjungan;
  final String keluhan;
  final String diagnosa;
  final String tekananDarah;
  final String suhuTubuh;
  final String beratBadan;
  final String tinggiBadan;

  RekapMedisModel({
    required this.tanggalKunjungan,
    required this.keluhan,
    required this.diagnosa,
    required this.tekananDarah,
    required this.suhuTubuh,
    required this.beratBadan,
    required this.tinggiBadan,
  });

  factory RekapMedisModel.fromJson(Map<String, dynamic> json) {
    return RekapMedisModel(
      tanggalKunjungan: json['tanggalKunjungan'],
      keluhan: json['keluhan'],
      diagnosa: json['diagnosa'],
      tekananDarah: json['tekananDarah'],
      suhuTubuh: json['suhuTubuh'],
      beratBadan: json['beratBadan'],
      tinggiBadan: json['tinggiBadan'],
    );
  }
}
